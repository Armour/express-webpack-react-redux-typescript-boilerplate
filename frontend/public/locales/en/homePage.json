{
  "title": "Home Page",
  "carousel": {
    "focusButtonText": "Focus me!",
    "firstPanelTitle": "First Panel",
    "secondPanelTitle": "Second Panel",
    "thirdPanelTitle": "Third Panel",
    "fourthPanelTitle": "Fourth Panel",
    "firstPanelDescription": "This is your first panel, try to swipe it!",
    "secondPanelDescription": "This is your second panel, try to swipe it!",
    "thirdPanelDescription": "This is your third panel, try to swipe it!",
    "fourthPanelDescription": "This is your fourth panel, try to swipe it!",
    "tooltipText": "Click me! >. <",
    "toastText": "I am a toast!"
  },
  "pushpin": {
    "green": "Green",
    "blue": "Blue",
    "orange": "Orange",
    "red": "Red",
    "purple": "Purple",
    "cyan": "Cyan",
    "linkOne": "Link 1",
    "linkTwo": "Link 2"
  }
}
