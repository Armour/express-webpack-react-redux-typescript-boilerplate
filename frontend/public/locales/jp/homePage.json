{
  "title": "ホームページ",
  "carousel": {
    "focusButtonText": "フォーカス!",
    "firstPanelTitle": "最初のパネル",
    "secondPanelTitle": "第2のパネル",
    "thirdPanelTitle": "第3のパネル",
    "fourthPanelTitle": "第4のパネル",
    "firstPanelDescription": "これはあなたの最初のパネルです、それをスワイプしよう！",
    "secondPanelDescription": "これはあなたの第2のパネルです、それをスワイプしよう！",
    "thirdPanelDescription": "これはあなたの第3のパネルです、それをスワイプしよう！",
    "fourthPanelDescription": "これはあなたの第4のパネルです、それをスワイプしよう！",
    "tooltipText": "クリック! >. <",
    "toastText": "私はトーストです！"
  },
  "pushpin": {
    "green": "緑",
    "blue": "青",
    "orange": "橙",
    "red": "赤",
    "purple": "紫",
    "cyan": "シアン",
    "linkOne": "リンク 1",
    "linkTwo": "リンク 2"
  }
}
