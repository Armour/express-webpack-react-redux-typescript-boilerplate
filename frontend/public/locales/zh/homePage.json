{
  "title": "首页",
  "carousel": {
    "focusButtonText": "看我看我!",
    "firstPanelTitle": "第一个面板",
    "secondPanelTitle": "第二个面板",
    "thirdPanelTitle": "第三个面板",
    "fourthPanelTitle": "第四个面板",
    "firstPanelDescription": "这是你的第一个面板，试试左右滑动它！",
    "secondPanelDescription": "这是你的第二个面板，试试左右滑动它！",
    "thirdPanelDescription": "这是你的第三个面板，试试左右滑动它！",
    "fourthPanelDescription": "这是你的第四个面板，试试左右滑动它！",
    "tooltipText": "戳我! >. <",
    "toastText": "哇我是一个toast!"
  },
  "pushpin": {
    "green": "绿色",
    "blue": "蓝色",
    "orange": "橘色",
    "red": "红色",
    "purple": "紫色",
    "cyan": "青色",
    "linkOne": "链接 1",
    "linkTwo": "链接 2"
  }
}
