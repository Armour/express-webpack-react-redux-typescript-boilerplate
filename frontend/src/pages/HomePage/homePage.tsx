import React, { Fragment } from 'react';
import { Translation } from 'react-i18next';

import Carousel from './components/Carousel';
import Pushpin from './components/Pushpin';
import TranslationButton from './components/TranslationButton';

const styles = require('./homePage.scss');

export class HomePage extends React.Component {
  public componentDidMount() {
    document.querySelectorAll('.pushpin-nav').forEach((elem, _) => {
      const target = document.querySelector('.' + elem.getAttribute('data-target')!);
      const rect = target!.getBoundingClientRect();
      let top = rect.top;
      // Make sure element is not hidden (display: none) or disconnected
      if (rect.width || rect.height || target!.getClientRects().length) {
        top += window.pageYOffset - target!.ownerDocument!.documentElement!.clientTop;
      }
      const bottom = top + elem.parentElement!.getBoundingClientRect().height - rect.height;
      M.Pushpin.init(elem, {
        top,
        bottom,
      });
    });
  }

  public render() {
    return (
      <Translation ns='homePage'>
        {(t) => (
          <Fragment>
            <div className={styles['home-page-block']}>
              <h1 className='page-title'>{t('title')}</h1>
              <div className='container'>
                <Carousel />
              </div>
            </div>
            <Pushpin color='blue' depth='lighten-1' />
            <Pushpin color='green' depth='lighten-1' />
            <Pushpin color='orange' depth='lighten-1' />
            <Pushpin color='red' depth='lighten-1' />
            <Pushpin color='purple' depth='lighten-1' />
            <Pushpin color='cyan' depth='lighten-1' />
            <TranslationButton />
          </Fragment>
        )}
      </Translation>
    );
  }
}

export default HomePage;
