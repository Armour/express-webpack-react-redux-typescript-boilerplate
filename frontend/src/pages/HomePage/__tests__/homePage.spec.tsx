import 'materialize-css';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import { HomePage } from '../homePage';

jest.mock('../components/Carousel', () => 'Carousel');
jest.mock('../components/Pushpin', () => 'Pushpin');
jest.mock('../components/TranslationButton', () => 'TranslationButton');

describe('HomePage', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <HomePage />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
