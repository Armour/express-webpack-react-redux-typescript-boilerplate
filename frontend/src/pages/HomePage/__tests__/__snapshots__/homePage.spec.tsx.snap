// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should renders correctly 1`] = `
Array [
  <div
    className="home-page-block"
  >
    <h1
      className="page-title"
    >
      title
    </h1>
    <div
      className="container"
    >
      <Carousel />
    </div>
  </div>,
  <Pushpin
    color="blue"
    depth="lighten-1"
  />,
  <Pushpin
    color="green"
    depth="lighten-1"
  />,
  <Pushpin
    color="orange"
    depth="lighten-1"
  />,
  <Pushpin
    color="red"
    depth="lighten-1"
  />,
  <Pushpin
    color="purple"
    depth="lighten-1"
  />,
  <Pushpin
    color="cyan"
    depth="lighten-1"
  />,
  <TranslationButton />,
]
`;
